File: elixir/rotational-cipher/rotational_cipher.exs
defmodule RotationalCipher do
  @doc """
  Given a plaintext and amount to shift by, return a rotated string.

  Example:
  iex> RotationalCipher.rotate("Attack at dawn", 13)
  "Nggnpx ng qnja"
  """
  @spec rotate(text :: String.t(), shift :: integer) :: String.t()
  def rotate(text, shift) do
    for <<c <- text>>, into: "" do
      if Regex.match?(~r/[a-zA-Z]/, <<c>>) do
        <<rem(c + shift, 26) + ?z>>
      else
        <<c>>
      end
    end
  end
end

File: csharp/rotational-cipher/RotationalCipher.cs
using System.Text;

public static class RotationalCipher
{
    public static string Rotate(string text, int shiftKey)
    {
        StringBuilder result = new(text.Length);
        foreach (var c in text)
        {
            if (!char.IsLetter(c)) {result.Append(c);continue;}
            var d = char.IsUpper(c) ? 'A' : 'a';
            result.Append((char)((((c + shiftKey) - d) % 26) + d));
        }
        return result.ToString();
    }
}